RCTMapboxGL/RCTMapboxGLManager.m

```diff
@@ -186,9 +186,39 @@ - (NSDictionary *)customDirectEventTypes
                             
                             RCTMGLAnnotation *pin = [[RCTMGLAnnotation alloc] initWithLocationRightCallout:CLLocationCoordinate2DMake(coordinate.latitude, coordinate.longitude) title:title subtitle:subtitle id:id rightCalloutAccessory:imageButton];
                             [pins addObject:pin];
+                            
+                            if ([anObject objectForKey:@"annotationImage"]) {
+                                NSObject *annotationImage = [anObject valueForKey:@"annotationImage"];
+                                NSString *annotationImageURL = [annotationImage valueForKey:@"url"];
+                                CGFloat height = (CGFloat)[[annotationImage valueForKey:@"height"] floatValue];
+                                CGFloat width = (CGFloat)[[annotationImage valueForKey:@"width"] floatValue];
+                                if (!height || !width) {
```

Comment:
You can get a UIImage’s size from the [`size` property](https://developer.apple.com/library/ios/documentation/UIKit/Reference/UIImage_Class/index.html#//apple_ref/occ/instp/UIImage/size). I suppose you can keep these properties to let the developer resize the image.